### PR TITLE
xz-bootstrap: Remove nonessential features

### PR DESCRIPTION
**Pull request by ryandesign**
#### Description

Don't link with gettext.
Don't use libiconv.
Don't create a static library.
Don't create lzmautils compatibility programs and symlinks.
Don't create the xzdec program.
Don't create the scripts.
Don't install documentation.
Don't install manpages.

See: https://trac.macports.org/ticket/58526

###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on
macOS 10.13.6 17G12034
Xcode 9.4.1 9F2000

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by jmroot** [comment]
What does xz do with libiconv exactly? Does it have any effect on character set handling when extracting archives? Though I guess it doesn't matter as long as it still works on the distfiles of the handful of ports that depend on xz-bootstrap.

BTW, the my_prefix stuff is no longer necessary since macports/macports-base@d840f25.

**Comment by ryandesign** [comment]
> What does xz do with libiconv exactly?

I'm not sure. Maybe nothing. xz doesn't link with libiconv, even before this change. It may be there solely to support gettext. xz's m4/gettext.m4 has a section beginning:

```
  dnl Sometimes libintl requires libiconv, so first search for libiconv.
```

The dependencies on gettext and libiconv have both been there ever since the port was originally added in 2cb1583888e249106baae183bc8a3c70a5cac7c4 as lzmautils-devel back in 2007. That was before we fixed [the libtool overlinking problem](https://trac.macports.org/ticket/38010) in MacPorts 2.2.0 in 2013. Presumably before then xz did link with libiconv unnecessarily. Perhaps now the dependency should be removed entirely even in xz but it won't really affect users since gettext still depends on libiconv.

> BTW, the my_prefix stuff is no longer necessary since [macports/macports-base@d840f25](https://github.com/macports/macports-base/commit/d840f25).

So basically 8e08962ba955d1743b6ac30a46dbef3ec922a7d5 should be reverted?

**Comment by jmroot** [comment]
> So basically [8e08962](https://github.com/macports/macports-ports/commit/8e08962ba955d1743b6ac30a46dbef3ec922a7d5) should be reverted?

Yes.

**Comment by kencu** [comment]
From the above comments, I don't exactly see what has to be done before this can be committed.
